patches/react-native-share-menu+6.0.0.patch

```diff
@@ -0,0 +1,234 @@
++        guard let userDefaults = UserDefaults(suiteName: "group.com.chat.expensify.chat") else {
+             print("Error: \(NO_APP_GROUP_ERROR)")
+             return
+         }
+@@ -102,7 +101,7 @@ class ShareMenu: RCTEventEmitter {
+     func getSharedText(callback: RCTResponseSenderBlock) {
+         var data = [DATA_KEY: sharedData] as [String: Any]
+ 
+-        if let bundleId = Bundle.main.bundleIdentifier, let userDefaults = UserDefaults(suiteName: "group.\(bundleId)") {
++        if let bundleId = Bundle.main.bundleIdentifier, let userDefaults = UserDefaults(suiteName: "group.com.chat.expensify.chat") {
+             data[EXTRA_DATA_KEY] = userDefaults.object(forKey: USER_DEFAULTS_EXTRA_DATA_KEY) as? [String: Any]
+         } else {
+             print("Error: \(NO_APP_GROUP_ERROR)")
+diff --git a/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift b/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift
+index e290cce..7b1c88b 100644
+--- a/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift
++++ b/node_modules/react-native-share-menu/ios/Modules/ShareMenuReactView.swift
+@@ -18,8 +18,6 @@ public class ShareMenuReactView: NSObject {
+     }
+ 
+     public static func attachViewDelegate(_ delegate: ReactShareViewDelegate!) {
+-        guard (ShareMenuReactView.viewDelegate == nil) else { return }
+-
+         ShareMenuReactView.viewDelegate = delegate
+     }
+ 
+@@ -27,6 +25,11 @@ public class ShareMenuReactView: NSObject {
+         ShareMenuReactView.viewDelegate = nil
+     }
+ 
++    @objc
++    func constantsToExport() -> [String: Any]! {
++        return ["isExtension": ShareMenuReactView.viewDelegate?.loadExtensionContext() != nil]
++    }
++
+     @objc(dismissExtension:)
+     func dismissExtension(_ error: String?) {
+         guard let extensionContext = ShareMenuReactView.viewDelegate?.loadExtensionContext() else {
+@@ -45,6 +48,7 @@ public class ShareMenuReactView: NSObject {
+         }
+ 
+         extensionContext.completeRequest(returningItems: [], completionHandler: nil)
++        ShareMenuReactView.detachViewDelegate()
+     }
+ 
+     @objc
+@@ -55,6 +59,7 @@ public class ShareMenuReactView: NSObject {
+         }
+ 
+         viewDelegate.openApp()
++        ShareMenuReactView.detachViewDelegate()
+     }
+ 
+     @objc(continueInApp:)
+@@ -72,6 +77,7 @@ public class ShareMenuReactView: NSObject {
+         }
+ 
+         viewDelegate.continueInApp(with: items, and: extraData)
++        ShareMenuReactView.detachViewDelegate()
+     }
+ 
+     @objc(data:reject:)
+@@ -106,7 +112,17 @@ public class ShareMenuReactView: NSObject {
+                 }
+ 
+                 for provider in attachments {
+-                    if provider.hasItemConformingToTypeIdentifier(kUTTypeURL as String) {
++                    
++                    if provider.hasItemConformingToTypeIdentifier(kUTTypeFileURL as String) {
```

Comment:
Okay, what was happening with the crash was that for some reason, the patch didn't get applied to my copy of this file and I was missing this clause. Looking into why that is.